### PR TITLE
Issue #13869: Improve XdocGenerator to generate file in temp folder a…

### PR DESCRIPTION
**Pull request by AayushSaini101**
Related: https://github.com/checkstyle/checkstyle/issues/13869

PR with all CIs: https://github.com/checkstyle/checkstyle/pull/13929 , all green.

### CODE REVIEW

**Comment by AayushSaini101** [comment]
> lease contact support.
Can you please highlight which contact support? Github or CI

**Comment by romani** [comment]
Circle CI

**Comment by nrmancuso** [comment]
Why/how did @Kevin222004 author the commit in this PR?

**Comment by AayushSaini101** [comment]
> Why/how did @Kevin222004 author the commit in this PR?

Resolved @nrmancuso

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java [hunk 1]
```diff
@@ -68,6 +70,11 @@ public static void generateXdocContent() throws Exception {
...
                 sink.close();
             }
+            final File outputFile = new File(pathToFile.replace(".template", ""));
+            final StandardCopyOption exp = StandardCopyOption.REPLACE_EXISTING;
+            Files.copy(tempFile.toPath(), outputFile.toPath(), exp);
+
         }
     }
 }
+
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java:80]: Please remove this extra line

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java [hunk 2]
```diff
@@ -68,6 +70,11 @@ public static void generateXdocContent() throws Exception {
             finally {
                 sink.close();
             }
+            final File outputFile = new File(pathToFile.replace(".template", ""));
+            final StandardCopyOption exp = StandardCopyOption.REPLACE_EXISTING;
+            Files.copy(tempFile.toPath(), outputFile.toPath(), exp);
+
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java:76]: Please remove this extra line

**Review by romani — CHANGES_REQUESTED** [review]
Last:

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java (hunk 2 above)
Comment (AayushSaini101) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java]: @romani Done

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java (hunk 1 above)
Comment (AayushSaini101) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java]: @romani Done

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
@nrmancuso , please finish review, I will run this update in PR created by me to make sure ci is passing.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java [hunk 3]
```diff
@@ -68,6 +70,9 @@ public static void generateXdocContent() throws Exception {
             finally {
                 sink.close();
             }
+            final File outputFile = new File(pathToFile.replace(".template", ""));
+            final StandardCopyOption exp = StandardCopyOption.REPLACE_EXISTING;
```
Comment (nrmancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java:74]:
```suggestion
            final StandardCopyOption copyOption = StandardCopyOption.REPLACE_EXISTING;
```

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Please provide evidence of testing that we delete temp file when we throw an exception during generation. Make some syntax error in a template file, show git diff, do `mvn clean verify`, show failure, list temp folder.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java (hunk 3 above)
Comment (AayushSaini101) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java:74]: Step 1: Added syntax error in the illegalcatch: 
image

Step 2: Git Status:
image

Step 3: Failed test case
]
Screenshot 2023-10-22 at 5 15 32 PM

ls in the temporary file 
image @romani @nrmancuso

Comment (AayushSaini101) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java]: > copyOption

@nrmancuso Done

**Review by nrmancuso — APPROVED** [review]

**Comment by nrmancuso** [comment]
@rnveach please finalize review since you had concerns in the issue

**Comment by rnveach** [comment]
My review was at https://github.com/checkstyle/checkstyle/issues/13869#issuecomment-1757690313 .

**Pull request merged**